### ISSUE TITLE
Blocked By

### ISSUE THREAD

**Issue by d2lam** [screwdriver-cd/screwdriver#474]
> As a user, I want my job to be blocked by other jobs/pipelines. 

Currently, there is no way to prevent a job from running the same time as some other job. A job should be able to be blocked by one or multiple jobs, from the same or different pipelines. 

Example:
```yaml
jobs: 
    main: ...
    test: ...
    deploy:
        blocked_by: [ test, sd@100, sd@150: staging ] 
```

The above `deploy` job would be blocked by `test` (currently, it will never run at the same time anyway since we don't have parallel build yet), or if any jobs in pipelineID `100` is running, or if the `staging` job in pipelineID `150` is running.

This should probably be built after we have the queuing infrastructure in place to make it easier.

**Comment by stjohnjohnson** [comment]
Closing in favor of https://github.com/screwdriver-cd/screwdriver/issues/653
